### DISCUSSION TITLE
2.3 to 2.4 Centos

### DISCUSSION THREAD

**Discussion by ng1991** [Security-Onion-Solutions/securityonion#13957] [General]
Hello,

I’ve been exploring options for upgrading our CentOS 7 installations, but I’m unsure of the best approach. We’re currently running Security Onion 2.3 on CentOS 7.

Is it possible to perform an in-place upgrade of Security Onion to version 2.4 using the ISO and then upgrade the OS to AlmaLinux? Or would a different approach be better to ensure data preservation?

I tested upgrading an unused forward node to AlmaLinux, but despite adding Salt and other adjustments, it doesn’t seem compatible with Security Onion 2.3.

Thank you, and I apologize if this has been addressed elsewhere.

I came across this discussion: https://github.com/Security-Onion-Solutions/securityonion/discussions/13116, but I’m not sure if it confirms whether Security Onion 2.4 supports CentOS 7.

**Comment by InfosecGoon** [comment]
CentOS 7 is end-of-life and is no longer receiving updates, security or otherwise.

Security Onion 2.4 is built on Oracle Linux 9. If you've deployed your Security Onion nodes with the /nsm volume on a separate physical device, it's possible to do an in-place upgrade that preserves the data; however, we don't really recommend this unless strictly necessary because it leaves a lot of loose ends to tie up (old indices not being included in data lifecycle management, that sort of thing).

Instructions are here: https://docs.securityonion.net/en/2.4/appendix.html

You could probably do something similar with migrating the contents of /nsm to a new network install on AlmaLinux on a new server, but the same caveats would apply.
